frontend/sw360-portlet/src/main/webapp/html/projects/view.jsp

```diff
@@ -321,17 +305,64 @@
                             }
                         }
                     ],
+                    "pageLength": 10,
                     "search": {smart: false},
                     "columns": [
                         {title: "Project Name", data: "name", render: {display: renderProjectNameLink}},
-                        {title: "Description", data: "description"},
-                        {title: "Project Responsible", data: "responsible"},
-                        {title: "State", data: "state"},
+                        {title: "Description", data: "desc", render: {display: renderDescription}},
+                        {title: "Project Responsible", data: "resp", render: {display: renderProjectResponsible}},
+                        {title: "State", data: "state", render: {display: renderStateBoxes}},
                         {title: "<span title=\"Release clearing state\">Clearing Status</span>", data: "clearing"},
                         {title: "Actions", data: "id", render: {display: renderProjectActions}}
                     ],
-                    "autoWidth": false
+                    "order": [[0, 'asc']],
+                    "lengthMenu": [[10, 25, 50, 100, -1], [10, 25, 50, 100, "All"]],
+                    "autoWidth": false,
+                    "fnDrawCallback": function (oSettings) {
+                        loadClearingStateSummaries();
+                    }
                 });
+
+                return projectsTable;
+            }
+
+            function renderStateBoxes(state, type, row) {
+                var projectStateBackgroundColour = '<%=PortalConstants.PROJECT_STATE_INACTIVE__CSS%>';
+                var clearingStateBackgroundColour = '<%=PortalConstants.CLEARING_STATE_UNKNOWN__CSS%>';
```

Comment:
Maybe extract the logic to another function `var clearingStateBackgroundColour = getClearingStateBackgroundColour(row)`

Reply:
added `getClearingStateBackgroundColour(row)`